packages/flutter/lib/src/material/material.dart

```diff
@@ -528,7 +528,8 @@ class _RenderInkFeatures extends RenderProxyBox implements MaterialInkController
   void _removeFeature(InkFeature feature) {
     assert(_inkFeatures != null);
     _inkFeatures!.remove(feature);
-    markNeedsPaint();
+    if (!feature._disposed)
```

Comment:
It seems odd that a feature is removed AFTER it has been disposed. Shouldn't that be the other way around?

Also, why are we skipping the call to `markNeedsPaint`? I'd expect that we need to update the painted content to not paint this feature anymore?

Reply:
Reverted, see https://github.com/flutter/flutter/pull/81445/files#r712709007